### PR TITLE
Make a start on docs

### PR DESCRIPTION
**Pull request by takluyver**
Narrative docs moved in from IPython repo and updated, API docs created anew.

### CODE REVIEW

**Comment by takluyver** [comment]
Docs now up at https://jupyter-client.readthedocs.org/

**Comment by rgbkrk** [comment]
This is great. Thank you so much @takluyver.

File: docs/api/client.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+client - communicating with kernels
+===================================
+
+.. module:: jupyter_client.client
```
Comment (minrk) [inline docs/api/client.rst]: I would put this as top-level in `jupyter_client`, so it's documented as `jupyter_client.KernelClient`. Same for Manager.

Comment (takluyver) [inline docs/api/client.rst]: Done in #23.

**Pull request merged**